src/service-object.ts

```diff
@@ -102,10 +100,14 @@ export interface DeleteCallback {
   (err: Error|null, apiResponse?: r.Response): void;
 }
 
-export interface GetConfig {
-  /**
-   * Create the object if it doesn't already exist.
-   */
+export interface CreateOptions {
+  // tslint:disable-next-line:no-any
+  [propertyName: string]: any;
+}
+
+export interface GetConfig extends CreateOptions {
+  // autoCreate:true creates the service object if it doesn't already exist
+  // in which case the rest of this config get passed into the "create" method
   autoCreate?: boolean;
 }
```

Comment:
The previous definition of `GetConfig` only allowed zero or one property (`autoCreate`) in the passed config object. In fact the "GetConfig" type should allow arbitrary additional parameters, which get passed into the the `ServiceObject`s 
 `createMethod`.

Reply:
I don't think this is entirely right though.  Every TypeScript project that uses this library should be implementing their own GetOptions interface for each object type.  The `GetConfig` only really exists because I guess `autoCreate` is available on every service-object.  I don't think having `GetConfig` extend basically `any` is the right thing here 🤷‍♂️